src/data_processing/packet_handler.rs

```diff
@@ -39,16 +38,13 @@ impl PacketHandler {
         self.clean_queue().await;
         // FIXME: Ideally this entire function should not lock the mutex all the time
 
-        match serde_json::from_str::<serde_json::Value>(new_message_string.as_str()) {
-            Ok(msg) => {
-                self.queue.lock().await.remove(&peer);
-                return Some(msg);
-            }
-            Err(_) => (),
+        if let Ok(msg) = serde_json::from_str(&new_message_string) {
```

Comment:
Since you were disposing of the error message, a full blown `match` block here wasn't really needed.

If you decide to stop disposing of the error from `serde_json::from_str()`, you'd go back to the match block.